docs/upgrade/money-class.md

```diff
@@ -0,0 +1,430 @@
+```
+
+## Extended and Custom Entities
+If you have created your custom entities that store monetary value or added a new such column [by entity extension](/docs/extensibility/entity-extension.md), change your `decimal` Doctrine type to the new `money` type:
+
+```diff
+  /**
+-  * @var string
+-  * @ORM\Column(type="decimal", precision=20, scale=6)
++  * @var \Shopsys\FrameworkBundle\Component\Money\Money
++  * @ORM\Column(type="money", precision=20, scale=6)
+   */
+  protected $myMonetaryValue;
+```
+
+If you do so, you'll have to create [a database migration](/docs/introduction/database-migrations.md) which will add a `(DC2Type:money)` comment on the columns, informing Doctrine about the specified type (similarly to the [`Version20190215092226`](/packages/framework/src/Migrations/Version20190215092226.php) in `shopsys/framework`).
+
+Because the entity property now changed the type, its usages (along with the related [entity data class](/docs/introduction/entities.md#entity-data)) have to be reviewed and changed as well.
+We recommend using type-hinting in the changed methods as explained in [Working With Money in PHP](#working-with-money-in-php) below.
+
+If you need to set a value to the property in these custom entities or data objects directly, you should use the `Money::create()` method that accepts a string or an integer (or another [construction method](/docs/introduction/how-to-work-with-money.md#construction)).
+
+For more details read [the Money in Doctrine section](/docs/introduction/how-to-work-with-money.md#money-in-doctrine) of the documentation.
+
+## Working With Money in PHP
+You should use `Money` for all monetary values across your project.
+We strongly recommend declaring the `Money` type for both [the method arguments](http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration) and [the return values](http://php.net/manual/en/functions.returning-values.php#functions.returning-values.type-declaration) in your methods.
+
+If you need to compute money, use its methods `add`, `subtract`, `multiply` and `divide` instead of the operators `+`, `-`, `*` and `/`.
+
+If you need to compare two instances of `Money`, use its methods `equals`, `isGreaterThan`, `isGreaterThanOrEqualTo`, `isLessThan`, `isLessThanOrEqualTo` and `compare` instead of the operators `==`/`===`, `>`, `>=`, `<`, `<=` and `<=>`.
+If you want to compare `Money` with zero, you can use the methods `isZero`, `isPositive` and `isNegative`.
+
+For more details read [the Money Class section](/docs/introduction/how-to-work-with-money.md#money-class) of the documentation.
+
+There were no changes needed in `shopsys/project-base` as most PHP implementation is in `shopsys/framework`.
+You might need to update your custom classes working with prices and overridden services, because of the changed types (and [strict typing](http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration.strict) in some cases).
```

Comment:
The note about stricted types deserves a dedicated section. It is important - it can cause fatal errors (or exceptions?) and here it looks like a subnote. And also stricted types are not directly related to `Money` class itself

Reply:
You're right, added in `declare(strict_types=1);`

Reply:
Please add a also note that phpstan will help as same as in "money in php"

Reply:
I didn't want to copy and I've realized that if you run phpstan, you'll get also errors for Strict typing, so you'll have to fix it together. I've tried to link it in 9182a1a97